packages/chevrotain/src/parse/exceptions_public.ts

```diff
@@ -21,55 +21,49 @@ export function isRecognitionException(error: Error) {
   return contains(RECOGNITION_EXCEPTION_NAMES, error.name)
 }
 
-export function MismatchedTokenException(
-  message: string,
-  token: IToken,
-  previousToken: IToken
-) {
-  this.name = MISMATCHED_TOKEN_EXCEPTION
-  this.message = message
-  this.token = token
-  this.previousToken = previousToken
-  this.resyncedTokens = []
-}
+abstract class RecognitionException
+  extends Error
+  implements IRecognitionException {
+  context: IRecognizerContext
+  resyncedTokens = []
 
-// must use the "Error.prototype" instead of "new Error"
-// because the stack trace points to where "new Error" was invoked"
-MismatchedTokenException.prototype = Error.prototype
+  protected constructor(message: string, public token: IToken) {
+    super(message)
 
-export function NoViableAltException(
-  message: string,
-  token: IToken,
-  previousToken: IToken
-) {
-  this.name = NO_VIABLE_ALT_EXCEPTION
-  this.message = message
-  this.token = token
-  this.previousToken = previousToken
-  this.resyncedTokens = []
-}
+    // fix prototype chain when typescript target is ES5
+    Object.setPrototypeOf(this, new.target.prototype)
 
-NoViableAltException.prototype = Error.prototype
+    /* istanbul ignore next - V8 workaround to remove constructor from stacktrace when typescript target is ES5 */
+    if (Error.captureStackTrace) {
```

Comment:
What would happen on Safari / Firefox ?

Reply:
Actually I think I now understand this workaround removes some redundant line from the stack trace.
Not fixes the whole stack trace, so its only a **minor** issue on none V8 runtimes.